File: NodeJsStmSrv/controller.js
const Materie = require("./Model/Materie");
const Nota = require("./Model/Nota");
const Student = require("./Model/Student");
const DBConnector = require("./Repository/DBConnector");

class Controller {

    constructor() {
        this.db = new DBConnector();
        this.db.connect();
    }

    addStudent(nume, prenume, grupa) {
        var student = new Student(nume, prenume, grupa);
        this.db.addStudent(nume, prenume, grupa);
    }

    editStudent(sID, newNume, newPrenume, newGrupa) {
        this.db.editStudent(sID, newNume, newPrenume, newGrupa);
    }

    remStudent(sID) {
        this.db.remStudent(sID);
    }

    addMaterie(denumire, profesor) {
        var materie = new Materie(denumire, profesor);
        this.addMaterie(denumire, profesor);
    }

    editMaterie(mID, newDenumire, newProfesor) {
        this.editMaterie(mID, newDenumire, newProfesor);
    }

    remMaterie(mID) {
        this.db.remMaterie(mID);
    }

    addNota(mark, descriere, sID, mID) {
        var nota = new Nota(mark, descriere, sID, mID);
        this.db.addNota(mark, descriere, sID, mID);
    }

    editNota(nID, newMark, newDescriere) {
        this.db.editNota(nID, newMark, newDescriere);
    }

    listCatalog() {
        return this.db.listCatalog();
    }

    getDBConnector() {
        return this.db;
    }
}

module.exports = Controller;

File: PythonStmSrv/controller.py
from Model.Materie import Materie
from Model.Nota import Nota
from Model.Student import Student
from Repository.DBConnector import DBConnector


class Controller:
    def __init__(self):
        self._dbconnector = DBConnector()
        self._dbconnector.connect()
        
    def __del__(self):
        self._dbconnector.disconnect()

    def addStudent(self, nume, prenume, grupa):
        student = Student(nume, prenume, grupa)
        self._dbconnector.addStudent(nume, prenume, grupa)
    
    def editStudent(self, sID, newNume, newPrenume, newGrupa):
        self._dbconnector.editStudent(sID, newNume, newPrenume, newGrupa)

    def remStudent(self, sID):
        self._dbconnector.remStudent(sID)

    def addMaterie(self, denumire, profesor):
        materie = Materie(denumire, profesor)
        self._dbconnector.addMaterie(denumire, profesor)

    def editMaterie(self, mID, newDenumire, newProfesor):
        self._dbconnector.editMaterie(mID, newDenumire, newProfesor)

    def remMaterie(self, mID):
        self._dbconnector.remMaterie(mID)

    def addNota(self, mark, descriere, sID, mID):
        nota = Nota(mark, descriere, sID, mID)
        self._dbconnector.addNota(mark, descriere, sID, mID)

    def editNota(self, nID, newMark, newDescriere):
        self._dbconnector.editNota(nID, newMark, newDescriere)

    def listCatalog(self):
        return self._dbconnector.listCatalog()

    def getDBConnector(self):
        return self._dbConnector
